awx/main/utils/plugins.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Ansible, Inc.
+# All Rights Reserved.
+
+"""
+This module contains the code responsible for extracting the lists of dynamically discovered plugins.
+"""
+
+from functools import cache
+
+
+@cache
+def discover_available_cloud_provider_plugin_names() -> list[str]:
+    """Return a list of cloud plugin names available in runtime.
+
+    The discovery result is cached since it does not change throughout
+    the life cycle of the server run.
+
+    :returns: List of plugin cloud names.
+    :rtype: list[str]
+    """
+    from awx.main.models.inventory import InventorySourceOptions
+
+    return list(InventorySourceOptions.injectors.keys())
+
+
+def compute_cloud_inventory_sources() -> dict[str, str]:
+    """Return a dictionary of cloud provider plugin names
+    available plus source control management.
+
+    :returns: Dictionary of plugin cloud names plus source control.
+    :rtype: dict[str, str]
+    """
+    # Get the list of plugin names
```

Comment:
Let's drop the comments before merging. They shouldn't repeat the code. If code is unobvious, rename functions and variables to make it relay the same message as this comment.

Reply:
Thank you, I will go through and be sure to remove these before I squash and flip to code review!

Reply:
These should all be removed now!